### PR TITLE
[PfemFluid] set local size for large wall elements

### PR DESCRIPTION
**Pull request by AFranci**
A wall size distance is defined and used to avoid fluid leakage in case of having big wall elements in the initial mesh. If the wall elements are too big with respect to the mean size of the mesh, locally an averaged mesh size is used in the PFEMFluid processes of alphaShape (select_mesh_elements_..), adding nodes (generate_new_nodes_...) and removing nodes (remove_nodes_...). A general cleaning is also done.

### CODE REVIEW

File: applications/DelaunayMeshingApplication/custom_utilities/mesher_utilities.cpp [hunk 1]
```diff
@@ -537,17 +536,17 @@ namespace Kratos
...
         }
 
-	    if(coplanar>extra){
+        if (coplanar > extra)
+        {
           numcoplanar++;
         }
 
-	    if(fabs(coplanar)<=ortho){
+        if (fabs(coplanar) <= ortho)
```
Comment (AlejandroCornejo) [inline applications/DelaunayMeshingApplication/custom_utilities/mesher_utilities.cpp]: better to use std::abs

File: applications/DelaunayMeshingApplication/custom_utilities/mesher_utilities.cpp [hunk 2]
```diff
@@ -760,7 +760,8 @@ namespace Kratos
       for (unsigned int j = i + 1; j < lpofa.size2(); ++j)
       {
         double projection = inner_prod(FaceNormals[i], FaceNormals[j]);
-	    if( fabs(projection) >= 0.99 ){
+        if (fabs(projection) >= 0.99)
```
Comment (AlejandroCornejo) [inline applications/DelaunayMeshingApplication/custom_utilities/mesher_utilities.cpp]: idem

File: applications/PfemFluidDynamicsApplication/custom_processes/compute_average_pfem_mesh_parameters_process.hpp [hunk 3]
```diff
@@ -226,12 +204,6 @@ namespace Kratos
     maxInternalPointRefiningBox[1] = RefiningBoxMaximumPoint[1] - mrRemesh.RefiningBoxMeshSize;
     maxInternalPointRefiningBox[2] = RefiningBoxMaximumPoint[2] - mrRemesh.RefiningBoxMeshSize;
 
-    // std::cout<<" RefiningBoxMinimumPoint "<<mrRemesh.RefiningBoxMinimumPoint <<std::endl;
-    // std::cout<<" minExternalPointRefiningBox "<<mrRemesh.RefiningBoxMinExternalPoint <<std::endl;
-    // std::cout<<" minInternalPointRefiningBox "<<mrRemesh.RefiningBoxMinInternalPoint <<std::endl;
-    // std::cout<<"     RefiningBoxMaximumPoint "<<mrRemesh.RefiningBoxMaximumPoint <<std::endl;
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_processes/compute_average_pfem_mesh_parameters_process.hpp:232]: so clean.. :)

File: applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp [hunk 4]
```diff
@@ -766,8 +771,20 @@ namespace Kratos
...
+				const double tolerance = 1.8;
+				if (ratio > tolerance)
+				{
+					meanMeshSize *= 0.5;
+					meanMeshSize += 0.5 * rigidWallMeshSize;
+				}
+			}
+
+			const double limitEdgeLength = 1.4 * meanMeshSize;
+			const double safetyCoefficient2D = 1.5;
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp:787]: i love those factors :)

File: applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp [hunk 5]
```diff
@@ -924,12 +940,12 @@ namespace Kratos
...
 										{
 											// std::cout << " the nodes has more or less the same position of a previously inserted node" << NewPositions[j][0] << " " << NewPositions[j][1] << " versus " << NewPosition[0] << " " << NewPosition[1] << std::endl;
-											suitableElement = false;
+											suitableElementForSecondAdd = false;
 										}
 									}
 								}
 
-								if (suitableElement == true)
+								if (suitableElementForSecondAdd == true)
```
Comment (AlejandroCornejo) [inline applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp]: you can use `if (suitableElementForSecondAdd ) {`

File: applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp (hunk 4 above)
Comment (AFranci) [reply applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp:787]: months of "research"...

File: applications/DelaunayMeshingApplication/custom_utilities/mesher_utilities.cpp (hunk 2 above)
Comment (AFranci) [reply applications/DelaunayMeshingApplication/custom_utilities/mesher_utilities.cpp]: done!

File: applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp (hunk 5 above)
Comment (AFranci) [reply applications/PfemFluidDynamicsApplication/custom_processes/generate_new_nodes_before_meshing_process.hpp]: done!

**Review by AlejandroCornejo — APPROVED** [review]

**Pull request merged**